### PR TITLE
Rework on calc_next_comm_period

### LINKED ISSUE #54: Bug found on Blheli_S calc_next_comm_timing_fast function
### Describe the issue

A bug was found in a function imported from Blheli_S: calc_next_comm_timing_fast
https://github.com/bitdump/BLHeli/issues/647

Please, update the code and test it @ 80K RPM

### Bluejay version

0.18

### ESC variant

any

### PWM frequency

Any

### DShot bitrate

Any

### Bidirectional DShot

Any

### FC firmware

Betaflight  4.4

### Motor size

Any

### Configurator debug log

_No response_

### PR DESCRIPTION
**Pull request by damosvil**
calc_next_comm_period routine has been improved.

Now, when motor is started it uses the same code path for all speeds (and the same 1/16 weighted average for all) making commutation timing more stable. Performance is just the same, because optimized code paths were executed after many if then if checks. Also code is way simpler to understand, and to maintain, because it uses two main paths: one for when motor is starting (1/4 weighted average) and a second one for when motor is started (1/16 weighted average).

### CODE REVIEW

**Comment by stylesuxx** [comment]
I set your other branch as base, so we only have the fix here to review ;-)

**Comment by stylesuxx** [comment]
Interesting. Does this change low RPM performance in any way or has it any other implications for #39? Or is it purely cosmetic and readability?

Also I liked the analysis you did in the related issue, you think you could add this in the code too? Minus the divisor of 8 part obviously.

File: Bluejay_Timing.asm [hunk 1]
```diff
@@ -257,59 +216,6 @@ calc_next_comm_15deg_set_min:
...
-
-    clr C
-    mov A, Temp3                    ; Subtract a fraction
-    subb    A, Temp5
-    mov Temp3, A
-    mov A, Temp4
-    subb    A, #0
-    mov Temp4, A
-
-    ; Note: Temp2 is assumed to be zero (approx. Comm_Period4x_H / 4)
```
Comment (damosvil) [inline Bluejay_Timing.asm:279]: This assumption was not correct. New commutation time H can be different to zero

**Comment by damosvil** [comment]
> Also I liked the analysis you did in the related issue, you think you could add this in the code too? Minus the divisor of 8 part obviously.

I added this to the restructuring branch

**Comment by damosvil** [comment]
> Interesting. Does this change low RPM performance in any way or has it any other implications for #39? Or is it purely cosmetic and readability?

This pr may improve that situation because it uses a single weighted mean instead of 3, depending on the situation, but until benchmarks are done I am not sure.

File: Bluejay_Timing.asm (hunk 1 above)
Comment (stylesuxx) [reply Bluejay_Timing.asm:279]: Perfect, thank you.

**Comment by stylesuxx** [comment]
> This pr may improve that situation because it uses a single weighted mean instead of 3, depending on the situation, but until benchmarks are done I am not sure.

Alright, so this is something we could maybe ask @ChrisRosser to benchmark for us on his setup, no?

**Comment by stylesuxx** [comment]
Superseeded by #73 - closed.

**Pull request closed without merging**